### PR TITLE
fix: pattern messages and vehicle modal

### PR DESCRIPTION
**Pull request by nameetjambla**
in these files I have done changes in pattern messages and fix vehicle modal

### CODE REVIEW

File: src/app/pages/fleet/drivers/add-driver/add-driver.component.ts [hunk 1]
```diff
@@ -855,28 +862,26 @@ export class AddDriverComponent implements OnInit, OnDestroy, CanComponentDeacti
...
         this.driverData.firstName = result.firstName;
         this.driverData.lastName = result.lastName;
-        this.driverData.DOB = _.isEmpty(result.DOB) ?  null : result.DOB;
-        this.driverData.startDate = _.isEmpty(result.startDate) ?  null : result.startDate;
-        this.driverData.terminationDate = _.isEmpty(result.terminationDate) ?  null : result.terminationDate;
-        this.driverData.contractStart = _.isEmpty(result.contractStart) ?  null : result.contractStart;
-        this.driverData.contractEnd = _.isEmpty(result.contractEnd) ?  null : result.contractEnd;
-        this.driverData.crossBorderDetails.fastExpiry = _.isEmpty(result.crossBorderDetails.fastExpiry) ? null : result.crossBorderDetails.fastExpiry;
-        this.driverData.licenceDetails.licenceExpiry = _.isEmpty(result.licenceDetails.licenceExpiry) ?  null : result.licenceDetails.licenceExpiry;
+        this.driverData.startDate = result.startDate;
```
Comment (kunalfleethawks) [inline src/app/pages/fleet/drivers/add-driver/add-driver.component.ts:881]: Why did you remove this code?

File: src/app/pages/fleet/vehicles/add-vehicle-new/add-vehicle-new.component.ts [hunk 2]
```diff
@@ -767,12 +766,12 @@ export class AddVehicleNewComponent implements OnInit {
...
         this.purchase = {
           purchaseVendorID: result.purchase.purchaseVendorID,
-          warrantyExpirationDate: _.isEmpty(result.purchase.warrantyExpirationDate) ? null : result.purchase.warrantyExpirationDate,
+          warrantyExpirationDate: result.purchase.warrantyExpirationDate,
           warrantyExpirationDateReminder: result.purchase.warrantyExpirationDateReminder,
           purchasePrice: result.purchase.purchasePrice,
           purchasePriceCurrency: result.purchase.purchasePriceCurrency,
           warrantyExpirationMeter: result.purchase.warrantyExpirationMeter,
-          purchaseDate: _.isEmpty(result.purchase.purchaseDate) ? null : result.purchase.purchaseDate,
+          purchaseDate: result.purchase.purchaseDate,
```
Comment (kunalfleethawks) [inline src/app/pages/fleet/vehicles/add-vehicle-new/add-vehicle-new.component.ts:774]: Why did you remove this code?

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]
Please address the review comments

**Comment by nameetjambla** [comment]
this branch will break the code so I requested not to merge.
